packages/cli/src/utils/scaffold.js

```diff
@@ -22,34 +39,64 @@ const variablePrefixes = {
 const getVariableName = (action, noun) =>
   action === 'resource'
     ? [noun, 'resource'].join(' ')
-    : [variablePrefixes[action], noun];
+    : [variablePrefixes[action], noun].join(' ');
 
-const createTemplateContext = (action, noun, includeComments) => {
+/**
+ * Produce a valid snake_case key from one or more nouns, and fix the
+ * inconsistent version numbers that come from _.snakeCase.
+ *
+ * @example
+ * nounToKey('Cool Contact V10') // cool_contact_v10
+ */
+const nounToKey = (noun) => _.snakeCase(noun).replace(/V_(\d+)$/gi, 'v$1');
+
+/**
+ * Create a context object to pass to the template
+ * @param {Object} options
+ * @param {ActionType} options.actionType - the action type
+ * @param {string} options.noun - the noun for the action
+ * @param {boolean} [options.includeIntroComments] - whether to include comments in the template
+ * @returns {TemplateContext}
+ */
+const createTemplateContext = ({
+  actionType,
+  noun,
+  includeIntroComments = false,
+}) => {
   // if noun is "Cool Contact"
   return {
-    ACTION: action, // trigger
-    ACTION_PLURAL: plural(action), // triggers
+    ACTION: actionType, // trigger
+    ACTION_PLURAL: plural(actionType), // triggers
 
-    VARIABLE: _.camelCase(getVariableName(action, noun)), // getContact, the variable that's imported
-    KEY: snakeCase(noun), // "cool_contact", the action key
+    VARIABLE: _.camelCase(getVariableName(actionType, noun)), // getContact, the variable that's imported
+    KEY: nounToKey(noun), // "cool_contact", the action key
     NOUN: noun
       .split(' ')
       .map((s) => _.capitalize(s))
       .join(' '), // "Cool Contact", the noun
     LOWER_NOUN: noun.toLowerCase(), // "cool contact", for use in comments
     // resources need an extra line for tests to "just run"
-    MAYBE_RESOURCE: action === 'resource' ? 'list.' : '',
-    INCLUDE_INTRO_COMMENTS: includeComments,
+    MAYBE_RESOURCE: actionType === 'resource' ? 'list.' : '',
+    INCLUDE_INTRO_COMMENTS: includeIntroComments,
   };
 };
 
-const writeTemplateFile = async (
-  actionType,
-  templateContext,
+/**
+ * @param {Object} options
+ * @param {TemplateType} options.templateType - the template to write
+ * @param {'js' | 'ts'} options.language - the language of the project
+ * @param {string} options.destinationPath - where to write the file
+ * @param {boolean} options.preventOverwrite - whether to prevent overwriting
+ * @param {TemplateContext} options.templateContext - the context for the template
+ */
+const writeTemplateFile = async ({
+  templateType,
```

Comment:
```suggestion
  actionType,
```

Same reason as above

Reply:
[See above](https://github.com/zapier/zapier-platform/pull/884#discussion_r1825439822), deliberate choice because of tests possible beyond just action types.